plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/m2e/core/ui/preferences/MavenArchetypesPreferencePage.java

```diff
@@ -0,0 +1,255 @@
+	 *            description of the edited catalog
+	 */
+	public void editLocalCatalog(String catalogName, String catalogFile, String description) {
+		editCatalog(catalogName, catalogFile, description, false);
+	}
+
+	/**
+	 * Edit remote catalog and verify.
+	 * 
+	 * @param catalogName
+	 *            URL of the catalog file
+	 * @param catalogFile
+	 *            name of the catalog file
+	 * @param description
+	 *            description of the edited catalog
+	 */
+	public void editRemoteCatalog(String catalogName, String catalogFile, String description) {
+		editCatalog(catalogName, catalogFile, description, true);
+	}
+
+	/**
+	 * Edit remote catalog.
+	 * 
+	 * @param catalogName
+	 *            URL of the catalog file
+	 * @param catalogFile
+	 *            name of the catalog file
+	 * @param description
+	 *            description of the edited catalog
+	 * @param verify
+	 *            true - verify remote catalog, false - do not verify remote
+	 *            catalog
+	 */
+	public void editRemoteCatalog(String catalogName, String catalogFile, String description, boolean verify) {
+		editCatalog(catalogName, catalogFile, description, true);
+	}
+
+	/**
+	 * Private method for selecting catalog by name.
+	 * 
+	 * @param catalogName
+	 *            URL of the catalog file
+	 * @param catalogFile
+	 *            name of the catalog file
+	 * @param description
+	 *            description of the edited catalog
+	 * @param verify
+	 *            true - verify remote catalog, false - do not verify remote
+	 *            catalog (only for remote catalogs)
+	 */
+	private void editCatalog(String catalogName, String catalogFile, String description, boolean verify) {
+		selectCatalog(catalogName);
+		new PushButton(EDIT_CATALOG).click();
+		new DefaultCombo().setText(catalogFile);
+		new LabeledText(CATALOG_DESCRIPTION).setText(description);
+		if (verify) {
+			log.info(verifyURL());
+		}
+		new OkButton().click();
+	}
+
+	/**
+	 * If catalog with given name is available, returns true, otherwise false.
+	 * 
+	 * @return true - catalog with given name exists, otherwise false
+	 */
+	public boolean containsCatalog(String catalogName) {
+		for (String name : getCatalogNames()) {
+			if (name.equals(catalogName)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Private method for editing catalogs.
+	 * 
+	 * @param catalogName
+	 *            name of the catalog file
+	 */
+	private void selectCatalog(String catalogName) {
+		for (TableItem item : getCatalogs()) {
+			if (item.getText().equals(catalogName)) {
+				item.select();
+			}
+		}
+	}
+
+	/**
+	 * Private method for verifying remote catalog.
+	 * 
+	 * @return result of verification
+	 */
+	private String verifyURL() {
+		new PushButton(VERIFY_BUTTON).click();
+		Text labeledText = new LabeledText(REMOTE_CATALOG_SHELL);
+		String text = labeledText.getText();
+		if (text.equals(EMPTY_CATALOG_MESSAGE)) {
+			throw new IllegalArgumentException(EMPTY_CATALOG_EXCEPTION);
```

Comment:
Do not throw an exception in such a method. We just want to know whether the url is valid or not.